### DISCUSSION TITLE
[Question] Clang versions to create a manylinux_2_28 compliant wheel?

### DISCUSSION THREAD

**Discussion by aadwived** [pypa/manylinux#1646] [Q&A]
The manylinux_2_28 release contains GCC 12, but does not contain any version of clang compiler.

Do we have a list of compatible clang versions to create manylinux_2_28 compliant wheel?

**Comment by aadwived** [comment]
@mayeut Can you kindly help with this query?

**Comment by mayeut** [comment]
We don't have a list of clang compilers.
There are some ideas around clang mentioned in the last comments of https://github.com/pypa/manylinux/issues/1012

**Reply by mayeut** [reply]
Some efforts have been made to use clang to build manylinux images and some PoC pull requests were opened in other repos, although those are building executables rather than python extensions:
- https://github.com/scikit-build/ninja-python-distributions/pull/320
- https://github.com/scikit-build/cmake-python-distributions/pull/632

The interest was focused on speeding up building on QEMU architectures so the clang used is not a full llvm toolchain (no libc++, no compiler-rt builtins) & I still don't know about compatibility issues when building python extensions (although I don't suspect there are any).
